build-tools/packages/build-cli/src/api.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable camelcase */
+import { Octokit } from "@octokit/core";
+import { Logger, defaultLogger } from "@fluidframework/build-tools";
+
+const OWNER = "microsoft";
+const REPO_NAME = "FluidFramework";
+const PULL_REQUEST_EXISTS = "GET /repos/{owner}/{repo}/pulls";
+const PULL_REQUEST_INFO = "GET /repos/{owner}/{repo}/commits/{commit_sha}/pulls";
+const PULL_REQUEST = "POST /repos/{owner}/{repo}/pulls";
+const ASSIGNEE = "POST /repos/{owner}/{repo}/issues/{issue_number}/assignees";
+const REVIEWER = "POST /repos/{owner}/{repo}/pulls/{pull_number}/requested_reviewers";
+const LABEL = "POST /repos/{owner}/{repo}/issues/{issue_number}/labels";
+const GET_USER = "GET /repos/{owner}/{repo}/branches/{branch}/protection/restrictions/users";
+const DESCRIPTION = `
+        ## Main-next integrate PR
+        The aim of this pull request is to sync main and next branch. The expectation from the assignee is as follows:
+        > - Acknowledge the pull request by adding a comment -- "Actively working on it".
+        > - Resolve any merge conflicts between this branch and next (and push the resolution to this branch). Merge next into this branch if needed. **Do NOT rebase or squash this branch: its history must be preserved**.
+        > - Ensure CI is passing for this PR, fixing any issues. Please don't look into resolving **Real service e2e test** and **Stress test** failures as they are **non-required** CI failures.
+        For more information about how to resolve merge conflicts and CI failures, visit [this wiki page](https://github.com/microsoft/FluidFramework/wiki/Main-next-Automation).`;
+const TITLE = "Automate: Main Next Integrate";
+const log: Logger = defaultLogger;
+
+/**
+ *
+ * @param token - GitHub authentication token
+ * @returns Returns true if pull request exists
+ */
+export async function pullRequestExists(token: string): Promise<boolean> {
+    log.verbose("CHECKING IF PULL REQUEST EXISTS");
```

Comment:
It's helpful to include what you're looking for in the log message. So if you're checking for  a PR with a title, include the title in the log message. It makes the logs more understandable on their own.